core/include/kernel/boot.h

```diff
@@ -69,8 +69,8 @@ void plat_cpu_reset_early(void);
 void plat_primary_init_early(void);
 unsigned long plat_get_aslr_seed(void);
 unsigned long plat_get_freq(void);
-#if defined(_CFG_CORE_STACK_PROTECTOR)
-uintptr_t plat_get_random_stack_canary(void);
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
+void plat_get_random_stack_canaries(void *buf, size_t blen, size_t clen);
```

Comment:
Please add an inline description, stating what `blen` and `clen` relate to. It should tell that the function is expected to panic upon error.

If this function is expected to always provide 2 32bit values, prefer:
`void plat_get_random_stack_canaries(uint32_t *start_canary, unint32_t *end_canary);`

Reply:
OK, I'll add an inline description for the function.
About the definition of `plat_get_random_stack_canaries`, it's defined to obtain canaries of variable number and individual size in byte. We hope this provides convenience for further expansion of other canary functions with different requirements.
Currently, thread canaries require two 32-bit canaries, while compiler canaries require a single 32- or 64-bit canary.